api_compatibility_policy.md

```diff
@@ -0,0 +1,85 @@
+# API compatibility policy
+
+This document proposes a policy regarding making API updates to the CRDs in this repo.
+Users should be able to build on the APIs in these projects with a clear idea of what
+they can rely on and what should be considered in progress and therefore likely to change.
+
+For these purposes the CRDs are divided into three groups:
+
+* [`Build` and `BuildTemplate`] - from https://github.com/knative/build
+* [`TaskRun`, `Task`, and `ClusterTask`] - "more stable"
+* [`PipelineRun`, `Pipeline` and `PipelineResource`] - "less stable"
+
+The use of `alpha`, `beta` and `GA` in this document is meant to correspond roughly to
+[the kubernetes API deprecation policies](https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-a-flag-or-cli).
+
+## What does compatibility mean here?
+
+This policy is about changes to the APIs of the [CRDs](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/),
+aka the spec of the CRD objects themselves.
+
+A backwards incompatible change would be a change the requires a user to update existing
+instances of these CRDs in clusters where they are deployed (after
+[automatic conversion is available](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definition-versioning/#webhook-conversion)
+this process may become less painful).
+
+The current process would have to look something like:
```

Comment:
*would have to look something like* seems a bit *too much* :upside_down_face: 

"would look something like" maybe ?

Reply:
haha good call!